services/QuillLMS/client/app/bundles/Evidence/components/studentView/container.tsx

```diff
@@ -441,27 +443,36 @@ export const StudentViewContainer = ({ dispatch, session, isTurk, location, acti
     refs[ref].current ? refs[ref].current.scrollIntoView(false) : null
   }
 
+  function clickedTextIsWithinInlineMarkupTag(event) {
+    return INLINE_MARKUP_TAGS.includes(event?.target?.nodeName)
+  }
+
   function handleHighlightKeyDown(e) {
     if (e.key !== 'Enter') { return }
-    toggleStudentHighlight(e.target.textContent)
+    toggleStudentHighlight(e)
   }
 
   function handleHighlightClick(e) {
-    toggleStudentHighlight(e.target.textContent)
+    toggleStudentHighlight(e)
   }
 
-  function toggleStudentHighlight(text, callback=null) {
-    const textWithHighlightContentRemoved = text.replace(STUDENT_HIGHLIGHT_STARTS_TEXT, '').replace(STUDENT_HIGHLIGHT_ENDS_TEXT, '')
+  function toggleStudentHighlight(eventOrString) {
+    if (typeof(eventOrString) === 'string') {
+      let stringText = eventOrString
+    } else {
+      let stringText = clickedTextIsWithinInlineMarkupTag(eventOrString) ? eventOrString.currentTarget.textContent : eventOrString.target.textContent
```

Comment:
Out of curiosity, what happens if we just always use `currentTarget` instead of `target`?

Reply:
I'm glad you asked that question - turns out we can use currentTarget in both code paths.